src/rest-server/src/models/v2/storage.js

```diff
@@ -82,6 +84,7 @@ const convertVolumeDetail = async (pvc) => {
       storage.type = 'other';
       storage.data = {};
     }
+    storage.readOnly = (pv.spec.flexVolume.readOnly === true);
```

Comment:
Seems we need to set PV as `readOnly` when we want to use readOnly feature.
But from the k8s API doc. The PV readOnly has different meaning. 
Refer to 
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/#flexpersistentvolumesource-v1-core `ReadOnly here will force the ReadOnly setting in VolumeMounts.` So it used to mount subpath as readOnly?

My suggestion is not storage readOnly in k8s object unless it has seem meaning. How about store it in PAI group object? If so, admin can config different read/write permission for same PVC.

Reply:
> ReadOnly here will force the ReadOnly setting in VolumeMounts

it's the same meaning, only k8s does not respect the field as said in docs yet, `spec.PersistentVolume.Spec.FlexVolume.ReadOnly` is never used when mounting currently

>  How about store it in PAI group object? If so, admin can config different read/write permission for same PVC.

it will conflict with pv spec if k8s respects the field in the future. in current implementation, admin can also configuration different storage permissions for different groups

Reply:
Actually, my concern is we really need to set is: `pod.spec.volumes.persistentVolumeClaim.readOnly` .

But now we use `spec.PersistentVolume.Spec.<StorageType>.ReadOnly`. 

It's two different settings and we can't make sure they will have same meaning for all storage type.
Seems only has a config for `pod.spec.volumes.persistentVolumeClaim.readOnly` is more make sense and less confuse.

Reply:
`Pod.Spec.Volumes.PersistentVolumeClaim.ReadOnly` is set at
https://github.com/microsoft/pai/blob/60ae3b54acc0b52ca9a77a8eb72914f25fac2810/src/rest-server/src/models/v2/job/k8s.js#L567-L573

`PersistentVolume.Spec.<PersistentVolumeSource>.ReadOnly` is used to tell whether the PV should be mounted as read only as k8s docs described